### PR TITLE
Escape special characters in email before creating a RegExp for case-…

### PR DESCRIPTION
**Pull request by alobodig**
…insensitive lookup

Fixes a bug with emails addresses containing special characters introduced after the fix for #1323. These characters were not being escaped and were interpreted as part of the regular expression.

### CODE REVIEW

**Comment by NullEntity** [comment]
If we're going to stay with the regex case insensitive search, at least anchor it to the start of the string by prefixing it with `^`. I'd personally prefer if all emails were just all converted to lower before being stored. Maybe it could be done as a pre save hook.

**Comment by alobodig** [comment]
Sticking with the regex also precludes the use of an index on the field (at least in mongo currently). I am definitely in favor of saving a lower case version instead.

**Comment by morenoh149** [comment]
double check travis ci

**Comment by JedWatson** [comment]
Saving a lowercase version isn't something we can easily do with backwards compatibility so this is a better option for now. Needs some fixes though, I'll comment inline.

Travis errors are unrelated linter errors, not sure why those are showing up for this PR !? -- is it up to date with master?

File: lib/session.js [hunk 1]
```diff
@@ -85,8 +85,11 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 	}
 	var User = keystone.list(keystone.get('user model'));
 	if ('string' === typeof lookup.email && 'string' === typeof lookup.password) {
+		// create regex for email lookup with special characters escaped
+		var emailRegExp = new RegExp('^' + lookup.email.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&') + '$', 'i');
```
Comment (JedWatson) [inline lib/session.js]: we have an `escapeRegExp` method in `keystone-utils`, use that please.

**Comment by NullEntity** [comment]
@JedWatson how about storing the current keystone version in the db? That'd allow for updating db schema when Keystone's updated and an update script can be triggered. This script could convert all emails to lower.

**Comment by alobodig** [comment]
Yes, it's up to date with master. Looks like there's a problem there too.

**Comment by JedWatson** [comment]
@alobodig I found the issues in mater and fixed them - thanks for the update!

**Comment by alobodig** [comment]
@JedWatson Excellent!

**Pull request merged**